lib/internal/modules/esm/translators.js

```diff
@@ -54,6 +54,13 @@ function getSource(url) {
     throw new ERR_INVALID_URL_SCHEME(['file', 'data']);
   }
 }
+exports.defaultGetSource = defaultGetSource;
+
+function defaultTransformSource({ url, format, source },
```

Comment:
Thinking about the destructuring upgrade path concern on this signature, how about - 

```js
defaultTransformSource(source, { url, format }, defaultTransformSource)`
```

Note we should never pass the loader itself to any of the hooks as this is not a public exposed API.